src/ray/raylet/node_manager.cc

```diff
@@ -2578,6 +2587,7 @@ void NodeManager::ForwardTask(
                      << node_id;
     task.OnSpillback()(node_id, node_info.node_manager_address(),
                        node_info.node_manager_port());
+    task_dependency_manager_.TaskCanceled(task.GetTaskSpecification().TaskId());
```

Comment:
Haha, it's kind of scary task dep manager is trying to do stuff with direct call tasks. Hopefully, Ion's new scheduler will solve some of these problems by letting us bypass the existing code for lease requests.

Reply:
Yeah...I think I can actually change this to disable the corresponding `TaskPending` calls for direct tasks. I think it's actually incorrect to do that (the raylet will think that the task is pending even if it's already finished a while ago).